extensions/amp-youtube/0.1/amp-youtube.js

```diff
@@ -52,15 +69,41 @@ class AmpYoutube extends AMP.BaseElement {
     this.applyFillContent(iframe);
     iframe.width = width;
     iframe.height = height;
+
     this.element.appendChild(iframe);
-    /** @private {?Element} */
+    /** @private {!Element} */
     this.iframe_ = iframe;
-    return loadPromise(iframe);
+    this.iframe_.id = this.frameId_;
+
+    return loadPromise(iframe).then(() => {
+      // Make sure the YT player is ready for this. For some reason YT player
```

Comment:
I assume this is specifically about `postMessage`? I'd recommend to start listening as soon as possible. The iframe doesn't know if anyone is listening or not. Do they send any message to parent such as "ready to listen"?